### DISCUSSION TITLE
`momentuHMMData` for non-spatial models

### DISCUSSION THREAD

**Discussion by RTbecard** [bmcclintock/momentuHMM#80] [Q&A] [answered]
I'm able to make non-spatial HMMs by just passing `NA` values to my coordinate columns with `prepData()`.  I initially tried to do this using the `momentuHMMData()` constructor, but I coulden't as this is not exported to the namespace for the user to access, even though its listed in the manual and the documentation implies that it can be used to make a non-spatial `momentuHMMData` object.

I was wondering if there is a reason that `momentuHMMData` has no `@export` flag.  I'm not aware of any way to make a non-spatial `momentuHMMData` object, even through the resulting models seem to work fine without spatial data.

Interested to hear your thoughts!

**Comment by bmcclintock** [comment, marked as answer]
The intended workflow in momentuHMM is to first use `prepData()` to check that everything is in order and return a `momentuHMMData` object that can be safely passed to other functions like `fitHMM()`. As detailed in the help file for `prepData()` and the non-spatial examples in the vignette, if one has non-spatial data then simply set `coordNames=NULL`.

I don't see why it should be necessary to access `momentuHMMData()` (hence it is not exported), but if one really wants to access an internal function then see [here](https://stat.ethz.ch/R-manual/R-devel/library/base/html/ns-dblcolon.html).

**Comment by RTbecard** [comment]
Ah, I see.  My bad.  `coordNames=NULL` was what I was missing here.

Thanks for the speedy reply!
